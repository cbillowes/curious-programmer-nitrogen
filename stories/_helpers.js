import React from 'react'
import Layout from '../src/components/layout'
import { SmallSpacer } from '../src/components/spacer'

const Page = ({ className, children }) => {
  return (
    <Layout
      className={className}
    >
      <SmallSpacer />
      {children}
    </Layout>
  )
}

export default Page