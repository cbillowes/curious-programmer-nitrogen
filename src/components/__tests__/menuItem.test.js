import React from "react"
import MenuItem from "../menuItem"
import { getSnapshot } from "./_helpers"

const toggleOnClick = function(e) {}

describe(`Menu item`, () => {
  it(`should be active`, () => {
    const component = (
      <MenuItem
        item={{ to: `/blog`, name: `Blog` }}
        active={true}
        toggleOnClick={toggleOnClick}
      />
    )
    const tree = getSnapshot(component)
    expect(tree).toMatchSnapshot()
  })

  it(`should be inactive`, () => {
    const component = (
      <MenuItem
        item={{ to: `/blog`, name: `Blog` }}
        active={false}
        toggleOnClick={toggleOnClick}
      />
    )
    const tree = getSnapshot(component)
    expect(tree).toMatchSnapshot()
  })
})
