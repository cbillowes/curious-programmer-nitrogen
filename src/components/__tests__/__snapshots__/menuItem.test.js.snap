// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu item should be active 1`] = `
<a
  className="menu-item active"
  href="/#"
  onClick={[Function]}
  onKeyUp={[Function]}
  role="button"
  tabIndex="0"
>
  Blog
</a>
`;

exports[`Menu item should be inactive 1`] = `
Array [
  " ",
  <a
    className="menu-item"
    href="/blog"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    Blog
  </a>,
  " ",
]
`;
